--- a/net/minecraft/world/CompoundContainer.java
+++ b/net/minecraft/world/CompoundContainer.java
@@ -2,10 +_,16 @@
 
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
+import org.bukkit.Location;
+import org.bukkit.craftbukkit.entity.CraftHumanEntity;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.inventory.InventoryHolder;
+
+import java.util.List;
 
 public class CompoundContainer implements Container {
-   private final Container f_18910_;
-   private final Container f_18911_;
+   public final Container f_18910_;
+   public final Container f_18911_;
 
    public CompoundContainer(Container p_18913_, Container p_18914_) {
       this.f_18910_ = p_18913_;
@@ -70,6 +_,41 @@
 
    public boolean m_7013_(int p_18934_, ItemStack p_18935_) {
       return p_18934_ >= this.f_18910_.m_6643_() ? this.f_18911_.m_7013_(p_18934_ - this.f_18910_.m_6643_(), p_18935_) : this.f_18910_.m_7013_(p_18934_, p_18935_);
+   }
+
+   @Override
+   public List<ItemStack> getContents() {
+      return List.of();
+   }
+
+   @Override
+   public void onOpen(CraftHumanEntity who) {
+
+   }
+
+   @Override
+   public void onClose(CraftHumanEntity who) {
+
+   }
+
+   @Override
+   public List<HumanEntity> getViewers() {
+      return List.of();
+   }
+
+   @Override
+   public InventoryHolder getOwner() {
+      return null;
+   }
+
+   @Override
+   public void setMaxStackSize(int size) {
+
+   }
+
+   @Override
+   public Location getLocation() {
+      return null;
    }
 
    public void m_6211_() {
