--- a/net/minecraft/world/level/storage/Convertable.java
+++ b/net/minecraft/world/level/storage/Convertable.java
@@ -61,6 +61,10 @@
 import net.minecraft.world.level.levelgen.presets.WorldPresets;
 import org.slf4j.Logger;
 
+// CraftBukkit start
+import net.minecraft.world.level.dimension.WorldDimension;
+// CraftBukkit end
+
 public class Convertable {
 
     static final Logger LOGGER = LogUtils.getLogger();
@@ -114,7 +118,7 @@
     }
 
     private static DataPackConfiguration readDataPackConfig(Dynamic<?> dynamic) {
-        DataResult dataresult = DataPackConfiguration.CODEC.parse(dynamic);
+        DataResult<DataPackConfiguration> dataresult = DataPackConfiguration.CODEC.parse(dynamic); // CraftBukkit - decompile error
         Logger logger = Convertable.LOGGER;
 
         Objects.requireNonNull(logger);
@@ -241,7 +245,11 @@
                 WorldSettings worldsettings = WorldSettings.parse(dynamic, datapackconfiguration);
                 Lifecycle lifecycle1 = ((Lifecycle) pair.getSecond()).add(lifecycle);
 
-                return WorldDataServer.parse(dynamic, datafixer, i, nbttagcompound2, worldsettings, levelversion, (GeneratorSettings) pair.getFirst(), lifecycle1);
+                // CraftBukkit start - Add PDC to world
+                WorldDataServer worldDataServer = WorldDataServer.parse(dynamic, datafixer, i, nbttagcompound2, worldsettings, levelversion, (GeneratorSettings) pair.getFirst(), lifecycle1);
+                worldDataServer.pdc = nbttagcompound1.get("BukkitValues");
+                return worldDataServer;
+                // CraftBukkit end
             } catch (Exception exception) {
                 Convertable.LOGGER.error("Exception reading {}", path, exception);
                 return null;
@@ -389,7 +415,7 @@
         }
 
         public Path getDimensionPath(ResourceKey<World> resourcekey) {
-            return DimensionManager.getStorageFolder(resourcekey, this.levelDirectory.path());
+            return getStorageFolder(this.levelDirectory.path(), this.dimensionType); // CraftBukkit
         }
 
         private void checkLock() {
