--- a/net/minecraft/world/level/chunk/IChunkAccess.java
+++ b/net/minecraft/world/level/chunk/IChunkAccess.java
@@ -256,10 +265,11 @@
 
     public void setUnsaved(boolean flag) {
         this.unsaved = flag;
+        if (!flag) this.persistentDataContainer.dirty(false); // CraftBukkit - SPIGOT-6814: chunk was saved, pdc is no longer dirty
     }
 
     public boolean isUnsaved() {
-        return this.unsaved;
+        return this.unsaved || this.persistentDataContainer.dirty(); // CraftBukkit - SPIGOT-6814: chunk is unsaved if pdc was mutated
     }
 
     public abstract ChunkStatus getStatus();
@@ -392,6 +402,27 @@
         }
     }
     public void fillBiomesFromNoise(BiomeResolver biomeresolver, Climate.Sampler climate_sampler) {
         ChunkCoordIntPair chunkcoordintpair = this.getPos();
         int i = QuartPos.fromBlock(chunkcoordintpair.getMinBlockX());
