--- a/net/minecraft/world/inventory/HopperMenu.java
+++ b/net/minecraft/world/inventory/HopperMenu.java
@@ -5,6 +_,7 @@
 import net.minecraft.world.entity.player.Inventory;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
+import org.bukkit.inventory.InventoryView;
 
 public class HopperMenu extends AbstractContainerMenu {
    public static final int f_150576_ = 5;
@@ -39,6 +_,11 @@
 
    public boolean m_6875_(Player p_39647_) {
       return this.f_39638_.m_6542_(p_39647_);
+   }
+
+   @Override
+   public InventoryView getBukkitView() {
+      return null;
    }
 
    public ItemStack m_7648_(Player p_39651_, int p_39652_) {
