--- a/net/minecraft/world/entity/animal/allay/Allay.java
+++ b/net/minecraft/world/entity/animal/allay/Allay.java
@@ -88,14 +_,23 @@
    private final VibrationListener.VibrationListenerConfig f_238787_;
    private final DynamicGameEventListener<Allay.JukeboxListener> f_238563_;
    private final SimpleContainer f_218303_ = new SimpleContainer(1);
-   private @Nullable BlockPos f_238682_;
-   private long f_238791_;
+   @Nullable
+   public BlockPos f_238682_;
+   public long f_238791_;
    private float f_218304_;
    private float f_218305_;
    private float f_238687_;
    private float f_238541_;
    private float f_238552_;
 
+   // BBSMC start
+   public boolean forceDancing = false; // CraftBukkit
+
+   public void setCanDuplicate(boolean canDuplicate) {
+      this.f_19804_.m_135381_(net.minecraft.world.entity.animal.allay.Allay.f_238802_, canDuplicate);
+   }
+   // BBSMC end
+
    public Allay(EntityType<? extends Allay> p_218310_, Level p_218311_) {
       super(p_218310_, p_218311_);
       this.f_21342_ = new FlyingMoveControl(this, 20, true);
@@ -273,7 +_,7 @@
       ItemStack itemstack = p_218361_.m_21120_(p_218362_);
       ItemStack itemstack1 = this.m_21120_(InteractionHand.MAIN_HAND);
       if (this.m_239559_() && this.m_239735_(itemstack) && this.m_218324_()) {
-         this.m_218376_();
+         this.duplicateAllay();
          this.f_19853_.m_7605_(this, (byte)18);
          this.f_19853_.m_6269_(p_218361_, this, SoundEvents.f_144243_, SoundSource.NEUTRAL, 2.0F, 1.0F);
          this.m_239358_(p_218361_, itemstack);
@@ -326,7 +_,7 @@
 
    public boolean m_7243_(ItemStack p_218387_) {
       ItemStack itemstack = this.m_21120_(InteractionHand.MAIN_HAND);
-      return !itemstack.m_41619_() && itemstack.m_41726_(p_218387_) && this.f_218303_.m_19183_(p_218387_) && this.f_19853_.m_46469_().m_46207_(GameRules.f_46132_);
+      return !itemstack.m_41619_() && itemstack.m_41726_(p_218387_) && this.f_218303_.m_19183_(p_218387_) && net.minecraftforge.event.ForgeEventFactory.getMobGriefingEvent(this.f_19853_, this);
    }
 
    protected void m_7581_(ItemEntity p_218359_) {
@@ -446,7 +_,7 @@
       return f_238776_.test(p_239736_);
    }
 
-   private void m_218376_() {
+   public Allay duplicateAllay() {
       Allay allay = EntityType.f_217014_.m_20615_(this.f_19853_);
       if (allay != null) {
          allay.m_20219_(this.m_20182_());
@@ -455,15 +_,15 @@
          this.m_239811_();
          this.f_19853_.m_7967_(allay);
       }
-
+      return allay;
    }
 
-   private void m_239811_() {
+   public void m_239811_() {
       this.f_238791_ = 6000L;
       this.f_19804_.m_135381_(f_238802_, false);
    }
 
-   private boolean m_218324_() {
+   public boolean m_218324_() {
       return this.f_19804_.m_135370_(f_238802_);
    }
 
