--- a/net/minecraft/world/entity/AreaEffectCloud.java
+++ b/net/minecraft/world/entity/AreaEffectCloud.java
@@ -39,19 +_,35 @@
    private static final EntityDataAccessor<ParticleOptions> f_19700_ = SynchedEntityData.m_135353_(AreaEffectCloud.class, EntityDataSerializers.f_135036_);
    private static final float f_146781_ = 32.0F;
    private Potion f_19701_ = Potions.f_43598_;
-   private final List<MobEffectInstance> f_19685_ = Lists.newArrayList();
+   public final List<MobEffectInstance> f_19685_ = Lists.newArrayList();
    private final Map<Entity, Integer> f_19686_ = Maps.newHashMap();
    private int f_19687_ = 600;
-   private int f_19688_ = 20;
-   private int f_19689_ = 20;
+   public int f_19688_ = 20;
+   public int f_19689_ = 20;
    private boolean f_19690_;
-   private int f_19691_;
-   private float f_19692_;
-   private float f_19693_;
+   public int f_19691_;
+   public float f_19692_;
+   public float f_19693_;
    @Nullable
    private LivingEntity f_19694_;
    @Nullable
    private UUID f_19695_;
+
+   // BBSMC start
+   public void refreshEffects() {
+      if (!this.f_19690_) {
+         this.m_20088_().m_135381_(net.minecraft.world.entity.AreaEffectCloud.f_19698_, net.minecraft.world.item.alchemy.PotionUtils.m_43564_(net.minecraft.world.item.alchemy.PotionUtils.m_43561_(this.f_19701_, this.f_19685_)));
+      }
+   }
+
+   public String getPotionType() {
+      return Registry.f_122828_.m_7981_(this.f_19701_).toString();
+   }
+
+   public void setPotionType(String string) {
+      m_19722_(net.minecraft.core.Registry.f_122828_.m_7745_(new net.minecraft.resources.ResourceLocation(string)));
+   }
+   // BBSMC end
 
    public AreaEffectCloud(EntityType<? extends AreaEffectCloud> p_19704_, Level p_19705_) {
       super(p_19704_, p_19705_);
